core/store/src/trie/trie_storage.rs

```diff
@@ -43,10 +46,20 @@ impl TrieCache {
                         guard.put(hash, value.into());
                     }
```

Comment:
```suggestion
                    } else {
                        metrics::SHARD_CACHE_TOO_LARGE.with_label_values(&labels).inc();
                    }
```
We need to inc this metric here as well - this is my bug from the first impl